src/robotFactory.js

```diff
@@ -1,10 +1,71 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x ? coords.x : 0,
+      y: coords.y ? coords.y : 0,
```

Comment:
This way it's more simple.
```suggestion
      x: coords.x || 0,
      y: coords.y || 0,
```